sdk/servicebus/Azure.Messaging.ServiceBus/src/Receiver/ServiceBusReceiver.cs

```diff
@@ -150,37 +174,42 @@ public class ServiceBusReceiver : IAsyncDisposable
         /// <param name="sessionId"></param>
         /// <param name="options">A set of options to apply when configuring the consumer.</param>
         ///
-        private ServiceBusReceiver(
+        internal ServiceBusReceiver(
             ServiceBusConnection connection,
             string entityName,
             bool isSessionEntity,
             string sessionId = default,
             ServiceBusReceiverOptions options = default)
         {
             Argument.AssertNotNull(connection, nameof(connection));
-            options ??= new ServiceBusReceiverOptions();
+            Argument.AssertNotNull(options, nameof(options));
+            Argument.AssertNotNull(options.RetryOptions, nameof(options.RetryOptions));
+            Argument.AssertNotNullOrWhiteSpace(entityName, nameof(entityName));
+            connection.ThrowIfClosed();
 
-            IsSessionReceiver = isSessionEntity;
+            Identifier = DiagnosticUtilities.GenerateIdentifier(entityName);
             _connection = connection;
-            RetryPolicy = options.RetryOptions.ToRetryPolicy();
+            _retryPolicy = options.RetryOptions.ToRetryPolicy();
             ReceiveMode = options.ReceiveMode;
             PrefetchCount = options.PrefetchCount;
             EntityName = entityName;
+            IsSessionReceiver = isSessionEntity;
             _innerReceiver = _connection.CreateTransportReceiver(
                 entityName: EntityName,
-                retryPolicy: RetryPolicy,
+                retryPolicy: _retryPolicy,
                 receiveMode: ReceiveMode,
-                prefetchCount: (uint) PrefetchCount,
+                prefetchCount: (uint)PrefetchCount,
```

Comment:
We need to cast to uint since the public type is an int, but the type expected by AMQP layer is uint. We have validation in the property getter that ensures it is >= 0.